tests/app/main/test_briefs.py

```diff
@@ -1912,6 +1912,44 @@ def test_supplier_details_shown(self, data_api_client):
         assert requirements_data.row(2).cell(0) == "Email address"
         assert requirements_data.row(2).cell(1) == "contact@big.com"
 
+    def test_analytics_and_messages_applied_on_first_submission(self, data_api_client):
+        """Go through submitting to edit_brief_response and the redirect to view_response_result. Assert messages."""
+        self.set_framework_and_eligibility_for_api_client(data_api_client)
+        data_api_client.get_brief.return_value = self.brief
+        # data_api_client.find_brief_responses.return_value = self.brief_responses
+        data_api_client.get_brief_response.return_value = self.brief_response()
+        data_api_client.update_brief_response.return_value = {'success': True}
+        data_api_client.find_brief_responses.side_effect = [
+            {'briefResponses': []},
```

Comment:
One in each view, the test does the whole process from the initial non existence of a brief response, it then does the submit and show result. 
So on the `edit` page we look for a brief response and don't find one so we submit a new brief response, then we end up on the result page where we get the brief response we just submitted to show the result of it.